docs/modules/data-structures/pages/vector-search-overview.adoc

```diff
@@ -0,0 +1,233 @@
+= VectorCollection data structure design
+:description: A Hazelcast vector database engine is a specialized type of database, which is optimized for storing, searching, and managing vector embeddings and additional metadata. You can include values in the metadata to provide filtering, additional processing, or analysis of vectors.
+:page-enterprise: true
+:page-beta: true
+
+{description}
+
+For further information on the design of the Vector Collection data structure, see xref:data-structures:vector-collections.adoc[].
+
+== Architecture
+The main components of vector search are illustrated in the following high-level diagram:
+
+image:vector-search-components.png[The high-level diagram of the main components]
+
+The components shown in the diagram are as follows:
+
+* *Clients*: methods and libraries for connecting and working with the vector database. Currently, two language clients are available: Java and Python.
+
+* *Collections*: a set consisting of one or several indexes and a one metadata storage.
+For further information about vector collection, see xref:data-structures:vector-collections.adoc[Vector Collection].
+
+* *Indexes*: named sets of vectors. Vectors within the same index must have the same dimensions and be compared using the same metric. For further information on indexes, see <<index, Index Overview>>.
+
+* *Metadata storage*: key-value storage for storing metadata values.
+The key and value can be any objects; for example, JSON, unstructured text, or java-serialized POJO.
+For more information about the available storage types, see xref:serialization:serialization.adoc[Serializing Objects and Classes]
+
+An overview of vector collection is illustrated below:
+
+image:vector-collection.png[Vector collection overview]
+
+
+A user-defined key, which is assigned during the addition of the vector to the collection, establishes a link between the vector and the metadata. Each key corresponds to a single vector from each index.
+
+=== Index
+
+Essentially, each index serves as a distinct vector space.
+In terms of storage, the index is a graph where each node represents a vector, and the edges are organized to optimize search efficiency.
+
+The index is based on the link:https://github.com/jbellis/jvector[JVector] library, which implements a link:https://github.com/Microsoft/DiskANN[DiskANN] algorithm for similarity search.
+
+=== Partitioning and Replication
+
+Each collection is partitioned and replicated based on the system's general partitioning rules. Data partitioning is carried out using the collection key.
+
+For further information on Hazelcast partitioning, see xref:data-partitioning.adoc[Data Partitioning and Replication].
```

Comment:
Sorry, one more - this link is not resolving, change is given below, that should sort it!

Reply:
Applied.